src/sage/misc/lazy_list.pyx

```diff
@@ -1058,7 +1058,11 @@ cdef class lazy_list_from_function(lazy_list_generic):
             step         1
         """
         while len(self.cache) <= i:
-            self.cache.append(self.callable(len(self.cache)))
+            try:
+                value = self.callable(len(self.cache))
+            except StopIteration:
+                return 1
+            self.cache.append(value)
```

Comment:
I am not sure why, but it seems to be intended as seen in the doctests of `lazy_list_from_function.__init__`:
```
sage: def f(n):
....:     if n >= 5: raise StopIteration
....:     return 5 - n
sage: list(lazy_list_from_function(f))
[5, 4, 3, 2, 1]
```

Before Cython 3 it works because StopIteration in generators (`lazy_list_generic.__iter__` here) simply stops the iteration, while in Cython 3 it becomes RuntimeError.

Relevant Cython changelog item:
> [PEP-479](https://www.python.org/dev/peps/pep-0479) (generator_stop) is now enabled by default with language level 3. (Github issue [#2580](https://github.com/cython/cython/issues/2580))